File: leetcode/src/00001_two_sum/solution.go
package leetcode

func twoSum(nums []int, target int) []int {
	hm := map[int]int{}
	for i, num := range nums {
		j, ok := hm[target-num]
		if ok {
			return []int{j, i}
		} else {
			hm[num] = i
		}
	}
	return []int{}
}

File: leetcode/src/00001_two_sum/solution.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hm = {}
        for i, num in enumerate(nums):
            j = hm.get(target - num)
            if j is not None:
                return [j, i]
            hm[num] = i
        return []
